### PR TITLE
release: new contract!

### PR DESCRIPTION
**Pull request by antoncoding**
## Release: New contract!
* Update to use new contract (customized underlying, new signing scheme)
* Fix contract EIP712 bug (didn't hash the bytes)
* Fetch participant counts!

### CODE REVIEW

File: web/src/hooks/transaction/useCheckInRun.ts [hunk 1]
```diff
@@ -25,15 +23,17 @@ const useCheckInRun = (fields: CheckInFields, onSuccess?: () => void) => {
...
-        `0x${fields.s.padStart(64, '0')}`,
+        numberToHex(fields.activityId).padEnd(32, '0'), //checkIn data, we use activityId
+        fields.signature,
       ],
     },
     {
       onError: () => {
         toast.error('Error checking in.');
+        console.log(
+          'numberToHex(fields.activityId).padEnd(32)',
```
Comment (ryanycw) [inline web/src/hooks/transaction/useCheckInRun.ts:34]:
```suggestion
          'checkIn data (activityId): ',
```

File: web/src/hooks/transaction/useCheckInRun.ts [hunk 2]
```diff
@@ -25,15 +23,17 @@ const useCheckInRun = (fields: CheckInFields, onSuccess?: () => void) => {
       functionName: 'checkIn',
       args: [
         fields.challengeId,
-        fields.timestamp,
-        fields.v,
-        `0x${fields.r.padStart(64, '0')}`,
-        `0x${fields.s.padStart(64, '0')}`,
+        numberToHex(fields.activityId).padEnd(32, '0'), //checkIn data, we use activityId
```
Comment (ryanycw) [inline web/src/hooks/transaction/useCheckInRun.ts:26]: Should we also adjust the CheckIn log in contract? Log this instead of timestamp?

**Review by ryanycw — APPROVED** [review]

File: web/src/hooks/transaction/useCheckInRun.ts (hunk 2 above)
Comment (antoncoding) [reply web/src/hooks/transaction/useCheckInRun.ts:26]: yes!

Comment (antoncoding) [reply web/src/hooks/transaction/useCheckInRun.ts:26]: will open another PR and fix it in the next release

**Pull request merged**